frontend/packages/dev-console/src/components/pipelineruns/logs/Logs.tsx

```diff
@@ -0,0 +1,127 @@
+
+type LogsProps = {
+  resource: PodKind;
+  resourceStatus: string;
+  container: ContainerSpec;
+  render: boolean;
+  autoScroll: boolean;
+  onComplete: (containerName: string) => void;
+};
+
+const Logs: React.FC<LogsProps> = ({
+  resource,
+  resourceStatus,
+  container,
+  onComplete,
+  render,
+  autoScroll,
+}) => {
+  const { name } = container;
+  const { kind, metadata = {} } = resource;
+  const { name: resName, namespace: resNamespace } = metadata;
+  const contentRef = React.useRef<HTMLDivElement>(null);
+  const [fetching, setFetching] = React.useState<boolean>(false);
+  const [error, setError] = React.useState();
+  const appendMessage = React.useCallback(
+    (blockContent) => {
+      const contentLines = blockContent.split('\n').filter((line) => !!line);
+      if (contentRef.current && contentLines.length >= 0) {
+        const elements = contentLines.map((content) => {
+          const customElement = document.createElement('div');
+          customElement.textContent = content;
+          return customElement;
+        });
+        elements.forEach((element) => {
+          contentRef.current.append(element);
+        });
+        const lastElement = elements[elements.length - 1];
+        if (render && lastElement && autoScroll) {
+          lastElement.scrollIntoView({ behavior: 'smooth', block: 'end' });
+        }
+      }
+    },
+    [autoScroll, render],
+  );
+
+  React.useEffect(() => {
+    let ws: WSFactory;
+    if (!fetching) {
+      const urlOpts = {
+        ns: resNamespace,
+        name: resName,
+        path: 'log',
+        queryParams: {
+          container: name,
+          follow: 'true',
+        },
+      };
+      const watchURL = resourceURL(modelFor(kind), urlOpts);
+      if (resourceStatus === LOG_SOURCE_TERMINATED) {
+        coFetchText(watchURL)
+          .then((res) => {
+            appendMessage(res);
+            onComplete(name);
+          })
+          .catch((err) => {
+            setError(err);
+            onComplete(name);
+          });
+      } else {
+        const wsOpts = {
+          host: 'auto',
+          path: watchURL,
+          subprotocols: ['base64.binary.k8s.io'],
+        };
+        ws = new WSFactory(watchURL, wsOpts);
+        ws.onmessage((msg) => {
+          const message = Base64.decode(msg);
+          appendMessage(message);
+        })
+          .onclose(() => {
+            onComplete(name);
+          })
+          .onerror((err) => {
+            setError(err);
+            onComplete(name);
+          });
+      }
+      setFetching(true);
+    }
+    return () => {
+      ws && ws.destroy();
+    };
+  }, [appendMessage, fetching, kind, name, onComplete, resName, resNamespace, resourceStatus]);
+
+  React.useEffect(() => {
+    if (contentRef.current && render && autoScroll) {
+      contentRef.current.scrollIntoView({ behavior: 'smooth', block: 'end' });
+    }
+  }, [autoScroll, render]);
+  return (
+    <>
+      {error && (
```

Comment:
Ah, I can now understand why you were confused lol... this is not what I meant by putting it above the logs. I meant under the header and above the logs. Currently line 120-121.

Reply:
What happened to this chunk of code @sahil143 ? Am I blind or is it removed?

Reply:
Oh, you moved it to the other component...

Reply:
Yeah, I don't think you can do this. If there is an error on a given container, we don't want to float that to the top of the log. Was there UX direction here?

Reply:
Discussed with Sahil, it was a confusion point around the term "header". He'll move the error back to the Log component and between the *Step Header* and the *Step Log Content* elements.

Reply:
Done 😄